osu.Game/Skinning/UserSkinComponentLookup.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Skinning
+{
+    /// <summary>
+    /// A lookup class which is only for internal use, and explicitly to get a user-level configuration.
+    /// </summary>
+    internal class UserSkinComponentLookup : ISkinComponentLookup
```

Comment:
And here I guess I have my answer to the previous thing I guess, or maybe not?

The problem I have here is that `UserSkinComponentLookup` is going to wrap an `ISkinComponentLookup`, which _can be a_ `GlobalSkinnableContainerLookup`, which again confuses me because I delineate the notion of _components_ and _containers of components_ as two notions that _will not cross_. If that is the case, then I'd argue the following renames should also happen:

- `ISkinComponentLookup` -> `ISkinLookup`
- `UserSkinComponentLookup` -> `UserSkinLookup`

because they encapsulate both lookups of _individual components_ as well as _containers of components_.

If I am _not_ correct in assuming that containers of components and components are separate notions, then I guess disregard all I said and you can probably merge this.

Reply:
> `ISkinComponentLookup` -> `ISkinLookup`

I think I agree with this in principle, just to remove N+1 keywords. However....

> `UserSkinComponentLookup` -> `UserSkinLookup`

I don't agree with this one because it sounds like "look up this thing from the user's skin specifically", like "ignore any argon/whatever base skin". Original naming I think is best for this one.

Also, `UserSkinComponentLookup` may not be just containers in the future. It depends on the path this takes but it may be something passed through to a `ISkin.GetConfiguration(UserSkinComponentLookup)` to configure `SkinnableDrawable`s (but not override them).  
THAT SAID, I will stress that this is up in the air, so it's just my own ramblings at this point.

Reply:
Suggestions thrown around for that second one: `UserConfiguredSkinLookup` `JsonSkinLookup`. I think I'd be alright with that, they don't necessarily have to be 1-to-1.

Reply:
If `UserSkinComponentLookup` is going to wrap an `ISkinComponentLookup` then I'm not sure it's an option to drop `Component` from only one. It's essentially a decorator pattern, so it should be transformative in nature, and not add any new qualifiers or whatever - otherwise something is just straight up not shaped right.

Probably fine to leave as is, I'm not hugely fussed at this stage.

Reply:
FWIW I also proposed dropping "Component" from everywhere, including `GetDrawableComponent`, but smoogi didn't agree with this so I dropped it for now.

Reply:
Adding my two cents here, dropping the term "Component" in these lookup classes makes me kinda anxious because it gives a more generalised name for the type than what it's actually supposed to be used for. As in, we can have `ISkinLookup`, but that only will look up drawables/containers, it does not look up textures/samples/config/etc.

We even have `ManiaSkinConfigurationLookup` existing right now, and having that next to what would be `ManiaSkinLookup` does not sound good to me at all.